app/src/components/market/market_buy.tsx

```diff
@@ -152,27 +153,19 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
       setStatus(Status.Loading)
       setMessage(`Buying ${formatBigNumber(tradedShares, collateral.decimals)} shares ...`)
 
-      const user = await provider.getSigner().getAddress()
+      // add 10% to the approved amount because there can be precision errors
+      // this can be improved if, instead of adding the 1% fee manually in the front, we use the `calcMarketFee`
+      // contract method and add it to the result of `calcNetCost` result
+      const costWithErrorMargin = cost.mul(11000).div(10000)
```

Comment:
Since this is sending an extra 10% of collateral to the CPK, but not all of it ends up being used, does this mean that the CPK will keep some tokens?

Did you test if the proxy's collateral balance after buying was 0?

Reply:
Yes, the CPK will keep some tokens, you can check this situation comparing this transactions:
1 tx: Balance of the cpk: 27.398
https://rinkeby.etherscan.io/address/0x48821d9891da0cebd713a163771f8e1f3154f606
![Screenshot_20200121_135634](https://user-images.githubusercontent.com/1144028/72825882-9ac9ec80-3c56-11ea-98ab-ec12a5ce03e8.png)

2 tx: Balance of the cpk: 27.631
https://rinkeby.etherscan.io/address/0x48821d9891da0cebd713a163771f8e1f3154f606
![Screenshot_20200121_135644](https://user-images.githubusercontent.com/1144028/72825898-a0273700-3c56-11ea-8b8a-90b382400763.png)

I think the best solution is to remove the  extra 10% amount of collateral  we send to the cpk.
If we remove this extra amount, the balance of the cpk will be 27.631
tx: https://rinkeby.etherscan.io/address/0x48821d9891da0cebd713a163771f8e1f3154f606
![Screenshot_20200121_140414](https://user-images.githubusercontent.com/1144028/72826073-eb414a00-3c56-11ea-831b-96669f277dd0.png)

Reply:
I remove the amount with cost, are you ok with this ?